quill-orientdb/src/main/scala/io/getquill/context/orientdb/OrientDBIdiom.scala

```diff
@@ -60,7 +60,7 @@ trait OrientDBIdiom extends Idiom {
         a.token
       case a @ (
         _: Function | _: FunctionApply | _: Dynamic | _: OptionOperation | _: Block |
-        _: Val | _: Ordering | _: QuotedReference
+        _: Val | _: Ordering | _: QuotedReference | _: TraversableOperation
         ) =>
         fail(s"Malformed query $a.")
```

Comment:
could you change the error to something like `Malformed or unsupported construct: $a`